users/views.py

```diff
@@ -1,21 +1,75 @@
-from .forms import CustomUserCreationForm
+from datetime import timedelta, datetime
 from django.urls import reverse_lazy
 from django.views import generic
 from django.shortcuts import render
+from django.template.loader import render_to_string
+from django.contrib.auth import views
 from .models import CustomUser
+from .forms import CustomUserCreationForm
+from gamestore.settings import ACCOUNT_ACTIVATION_URL
 
 
 class SignUpView(generic.CreateView):
     form_class = CustomUserCreationForm
     success_url = reverse_lazy('login')
     template_name = 'users/signup.html'
 
+    # TODO--------------
+    def form_valid(self):
+        self.object = self.get_form().save()
+        return render(self.request, 'registration/login.html', context={'nextstep': True})
+
+# TODO ----------------
+class CustomLoginView(views.LoginView):
+
+    def get(self):
+        context = super().get_context_data()
+        context['nextstep'] = False
+        return self.render_to_response(context)
+
+
+def account_activation(request, user_id, active_link_time):
+    user = CustomUser.objects.get(id=user_id)
+    delta = timedelta(hours=3)
+    active_link_time = float(active_link_time)
+    difference = datetime.now() - datetime.fromtimestamp(active_link_time)
```

Comment:
So, basically user can copy the link from email, change the timestamp in it and paste the modified link to the browser to avoid link time expiration, right?